atc/db/build_factory_test.go

```diff
@@ -68,7 +68,7 @@ var _ = Describe("BuildFactory", func() {
 			DescribeTable("completed and past the grace period",
 				func(status db.BuildStatus, matcher types.GomegaMatcher) {
 					//set grace period to 0 for this test
-					buildFactory = db.NewBuildFactory(dbConn, lockFactory, 0)
+					buildFactory = db.NewBuildFactory(dbConn, lockFactory, 0, 0)
```

Comment:
Could you add tests under "pipeline builds" for the failed grace period similar to this one? The diff is you could have a new buildFactory with failedGracePeriod set to 0 or 1 second.

So then you could test under following situations:
build failed and check interceptable right away
build fialed and check interceptable after 1s